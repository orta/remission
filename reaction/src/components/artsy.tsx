import * as PropTypes from "prop-types"
import * as React from "react"

/**
 * The Artsy specific props injected by the higher-order component produced by `ContextConsumer`.
 *
 * @see {@link ContextProvider}
 * @see {@link ContextConsumer}
 *
 * @todo Once this PR for `rest` https://github.com/Microsoft/TypeScript/pull/13470 lands weâ€™ll be able to not make this
 *       optional and simply remove it from the props that a component wrapped with the `ContextConsumer` HOC accepts.
 */
export interface ContextProps {
  /**
   * The currently signed-in user.
   */
  currentUser?: User
}

interface PrivateContextProps extends ContextProps {
  /**
   * This prop solely exists to verify that a consumer is nested inside a provider.
   */
  _isNestedInProvider: boolean
}

const ContextTypes: React.ValidationMap<PrivateContextProps> = {
  _isNestedInProvider: PropTypes.bool,
  currentUser: PropTypes.object,
}

/**
 * This component provides Artsy specific context props to components down the tree. Components that wish to consume
 * these props should be wrapped with the `ContextConsumer` higher-order component.
 *
 * @see {@link ContextConsumer}
 */
export class ContextProvider extends React.Component<ContextProps, null>
  implements React.ChildContextProvider<PrivateContextProps> {
  static childContextTypes = ContextTypes

  constructor(props) {
    if (React.Children.count(props.children) > 1) {
      throw new Error("A ContextProvider expects a single child.")
    }
    super(props)
  }

  getChildContext() {
    return {
      _isNestedInProvider: true,
      currentUser: this.props.currentUser,
    }
  }

  render() {
    return this.props.children as any
  }
}

/**
 * Wraps a component in a higher-order component that injects the Artsy specific context props into the given component.
 * Wrapped components are expected to be rendered in a tree that contains a `ContextProvider` component.
 *
 * The injected props are described in `ContextProps`.
 *
 * @see {@link ContextProvider}
 * @see {@link ContextProps}
 *
 * @param Component The component that needs Artsy specific context props.
 */
export function ContextConsumer<P>(
  Component: React.ComponentClass<P> | React.StatelessComponent<P>
): React.ComponentClass<P> {
  const name = Component.displayName || Component.name
  return class extends React.Component<P, void> {
    static contextTypes = ContextTypes
    static displayName = `Artsy(${name})`

    constructor(props, context: PrivateContextProps) {
      if (!context._isNestedInProvider) {
        const start = name || "A component"
        throw new Error(`${start}, which needs Artsy props, was not wrapped inside a ContextProvider component.`)
      }
      super(props, context)
    }

    render() {
      const currentUser = this.context.currentUser
      const props = Object.assign({ currentUser }, this.props)
      return <Component {...props} />
    }
  }
}
