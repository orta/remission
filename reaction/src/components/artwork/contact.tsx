import * as React from "react"
import * as Relay from "react-relay"
import TextLink from "../text_link"

export interface ContactProps extends RelayProps, React.HTMLProps<ArtworkContact> {}

export class ArtworkContact extends React.Component<ContactProps, null> {
  contactLine() {
    const { artwork } = this.props
    if (artwork.sale && artwork.sale.is_auction) {
      return this.auctionLine()
    } else if (artwork.is_inquireable) {
      return this.contactPartnerLine()
    }
  }

  auctionLine() {
    const { artwork } = this.props
    if (artwork.sale.is_live_open) {
      return <TextLink href={artwork.href} underline>Enter Live Auction</TextLink>
    } else if (artwork.sale.is_open) {
      const sa = artwork.sale_artwork
      const bids = sa.counts.bidder_positions
      if (bids > 0) {
        const s = bids > 1 ? "s" : ""
        return <span>{sa.highest_bid.display} ({bids} bid{s})</span>
      } else {
        return <span>{sa.opening_bid.display}</span>
      }
    } else if (artwork.sale.is_closed) {
      return <span>Auction closed</span>
    } else {
      return <span />
    }
  }

  contactPartnerLine() {
    return (
      <TextLink href={this.props.artwork.href} underline>
        Contact {this.props.artwork.partner.type.toLowerCase()}
      </TextLink>
    )
  }

  render() {
    return <div>{this.contactLine()}</div>
  }
}

export default Relay.createContainer(ArtworkContact, {
  fragments: {
    artwork: () => Relay.QL`
      fragment on Artwork {
        _id
        href
        is_inquireable
        sale {
          is_auction
          is_live_open
          is_open
          is_closed
        }
        partner(shallow: true) {
          type
        }
        sale_artwork {
          highest_bid {
            display
          }
          opening_bid {
            display
          }
          counts {
            bidder_positions
          }
        }
      }
    `,
  },
})

interface RelayProps {
  artwork: {
    _id: string
    href: string | null
    is_inquireable: boolean | null
    sale: {
      is_auction: boolean | null
      is_live_open: boolean | null
      is_open: boolean | null
      is_closed: boolean | null
    } | null
    partner: {
      type: string | null
    } | null
    sale_artwork: {
      highest_bid: {
        display: string | null
      } | null
      opening_bid: {
        display: string | null
      } | null
      counts: {
        bidder_positions: boolean | number | string | null
      } | null
    } | null
  }
}
