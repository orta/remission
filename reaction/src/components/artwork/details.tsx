import * as React from "react"
import * as Relay from "react-relay"
import TextLink from "../text_link"

import styled from "styled-components"

const TruncatedLine = styled.div`
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`

interface Props extends RelayProps, React.HTMLProps<ArtworkDetails> {
  showSaleLine: boolean
}

export class ArtworkDetails extends React.Component<Props, null> {
  static defaultProps = {
    showSaleLine: true,
  }

  artistLine() {
    const { cultural_maker, artists } = this.props.artwork

    if (cultural_maker) {
      return <TruncatedLine><strong>{cultural_maker}</strong></TruncatedLine>
    } else if (artists && artists.length) {
      const artistLine = artists
        .reduce((acc, artist) => {
          return acc.concat([", ", <TextLink href={artist.href} key={artist.__id}>{artist.name}</TextLink>])
        }, [])
        .slice(1)
      return <TruncatedLine><strong>{artistLine}</strong></TruncatedLine>
    }
  }

  titleLine() {
    return (
      <TruncatedLine>
        <TextLink href={this.props.artwork.href}>
          <em>{this.props.artwork.title}</em>
          {this.props.artwork.date && `, ${this.props.artwork.date}`}
        </TextLink>
      </TruncatedLine>
    )
  }

  partnerLine() {
    if (this.props.artwork.collecting_institution) {
      return <TruncatedLine>{this.props.artwork.collecting_institution}</TruncatedLine>
    } else if (this.props.artwork.partner) {
      return (
        <TruncatedLine>
          <TextLink href={this.props.artwork.partner.href}>
            {this.props.artwork.partner.name}
          </TextLink>
        </TruncatedLine>
      )
    }
  }

  saleLine() {
    const artwork = this.props.artwork
    const hasSaleMessage = artwork.sale_message && artwork.sale_message !== "Contact For Price"
    const notInAuction = !(artwork.sale && artwork.sale.is_auction)
    if (hasSaleMessage && notInAuction) {
      return <div>{artwork.sale_message}</div>
    }
  }

  render() {
    return (
      <div>
        {this.artistLine()}
        {this.titleLine()}
        {this.partnerLine()}
        {this.props.showSaleLine && this.saleLine()}
      </div>
    )
  }
}

export default Relay.createContainer(ArtworkDetails, {
  fragments: {
    artwork: () => Relay.QL`
      fragment on Artwork {
        href
        title
        date
        sale_message
        cultural_maker
        artists(shallow: true) {
          __id
          href
          name
        }
        collecting_institution
        partner(shallow: true) {
          name
          href
        }
        sale {
          is_auction
          is_live_open
          is_open
          is_closed
        }
      }
    `,
  },
})

interface RelayProps {
  artwork: {
    href: string | null
    title: string | null
    date: string | null
    sale_message: string | null
    cultural_maker: string | null
    artists: Array<{
      __id: string
      href: string | null
      name: string | null
    } | null> | null
    collecting_institution: string | null
    partner: {
      name: string | null
      href: string | null
    } | null
    sale: {
      is_auction: boolean | null
      is_live_open: boolean | null
      is_open: boolean | null
      is_closed: boolean | null
    } | null
  }
}
