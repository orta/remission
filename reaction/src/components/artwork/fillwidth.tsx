import * as React from "react"
import * as Relay from "react-relay"

import sizeMe from "react-sizeme"
import styled from "styled-components"
import fillwidthDimensions from "../../utils/fillwidth"
import Artwork from "./fillwidth_item"

import { find } from "lodash"

interface RelayProps {
  artworks: {
    edges: Array<{
      node: any
    } | null> | null
  }
}

interface Props extends RelayProps, React.HTMLAttributes<Fillwidth> {
  targetHeight?: number
  gutter?: number
  size?: any
}

export class Fillwidth extends React.Component<Props, null> {
  public static defaultProps: Partial<Props>

  renderArtwork(artwork, dimensions, i) {
    const { gutter } = this.props
    const artworkSize = find(dimensions, ["__id", artwork.__id])
    return (
      <Artwork
        artwork={artwork as any}
        key={"artwork--" + artwork.__id}
        targetHeight={artworkSize.height}
        imageHeight={artworkSize.height}
        width={artworkSize.width}
        margin={i === dimensions.length - 1 ? 0 : gutter}
      />
    )
  }

  render() {
    const artworks = this.props.artworks.edges
    const dimensions = fillwidthDimensions(
      this.props.artworks.edges,
      this.props.size.width,
      this.props.gutter,
      this.props.targetHeight
    )
    return (
      <div className={this.props.className}>
        {artworks.map((artwork, i) => this.renderArtwork(artwork.node, dimensions, i))}
      </div>
    )
  }
}

const StyledFillwidth = styled(Fillwidth)`
  margin-bottom: 50px;
`

StyledFillwidth.defaultProps = {
  targetHeight: 180,
  gutter: 10,
}

const ArtworkFragment = Relay.QL`
  fragment on Artwork {
    __id
    image {
      aspect_ratio
    }
    ${Artwork.getFragment("artwork")}
  }
`
const sizeMeOptions = {
  monitorHeight: false,
  refreshRate: 64,
  refreshMode: "debounce",
}

const FillwidthDimensions = sizeMe(sizeMeOptions)(StyledFillwidth) as React.StatelessComponent<Props>

export default Relay.createContainer(FillwidthDimensions, {
  fragments: {
    artworks: () => Relay.QL`
      fragment on ArtworkConnection {
        edges {
          node {
            ${ArtworkFragment}
          }
        }
      }
    `,
  },
})
