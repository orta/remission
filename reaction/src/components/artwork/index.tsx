import * as React from "react"
import styled, { css } from "styled-components"

import theme from "../../assets/theme"
import ArtworkMetadata, { ArtworkMetadataProps } from "./metadata"
import createContainer, { RelayProps } from "./relay"

const Container = styled.div`
  width: 100%;
`

const ImageContainer = styled.div`
  width: 100%;
  position: relative;
  cursor: pointer;

  &:before {
    content: '';
    display: block;
    padding-bottom: 100%;
  }

  & .overlay-container {
    transition: opacity 0.25s;

    &.hovered {
      opacity: 0;
      visibility: hidden;
    }

    &.selected {
      opacity: 1;
      visibility: visible;
    }
  }

  &:hover {
    & .overlay-container.hovered {
      ${css`
        @media (min-width: ${theme.flexboxgrid.breakpoints.sm + 1}px) {
          opacity: 1;
          visibility: visible;
        }
      `}
    }
  }
`

const Image = styled.img`
  max-width: 100%;
  max-height: 100%;
  margin: auto;
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
`

export interface OverlayProps {
  selected: boolean
}

export interface ArtworkProps extends RelayProps {
  size?: number
  extended?: boolean
  Overlay?: React.SFC<OverlayProps>
  onSelect?: (selected: boolean) => void
  showOverlayOnHover?: boolean
}

export interface ArtworkState {
  isSelected: boolean
}

export class Artwork extends React.Component<ArtworkProps, ArtworkState> {
  static defaultProps = {
    size: 250,
    extended: true,
    overlay: null,
    showOverlayOnHover: false,
  }

  constructor(props) {
    super(props)
    this.state = {
      isSelected: false,
    }
  }

  onSelected(e) {
    if (!this.props.Overlay) {
      return
    }

    this.setState({
      isSelected: !this.state.isSelected,
    })

    if (this.props.onSelect) {
      this.props.onSelect(!this.state.isSelected)
    }
  }

  render() {
    const { size, artwork, Overlay, showOverlayOnHover } = this.props
    let overlayClasses = "overlay-container"

    overlayClasses += showOverlayOnHover ? " hovered" : ""
    overlayClasses += this.state.isSelected ? " selected" : ""

    return (
      <Container onClick={this.onSelected.bind(this)} size={size}>
        <ImageContainer size={size}>
          <Image src={artwork.image.url} size={size} />
          <div className={overlayClasses}>
            {Overlay && <Overlay selected={this.state.isSelected} />}
          </div>
        </ImageContainer>
        <ArtworkMetadata artwork={artwork} />
      </Container>
    )
  }
}

export default createContainer<ArtworkProps, ArtworkMetadataProps>(Artwork, ArtworkMetadata)
