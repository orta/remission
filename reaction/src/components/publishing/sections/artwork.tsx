import React from "react"
import styled from "styled-components"
import { resize } from "../../../utils/resizer"
import { Layout, SectionLayout } from "../typings"
import ArtworkCaption from "./artwork_caption"
import ImageWrapper from "./image_wrapper"

interface ArtworkProps {
  artwork: any
  layout?: Layout
  sectionLayout?: SectionLayout
  linked?: boolean
  width?: string | number
  height?: string | number
}

const ArtworkImage: React.SFC<ArtworkProps> = props => {
  const { artwork, linked, height, width, layout } = props
  const src = resize(artwork.image, { width: 1200 })
  const image = (
    <ImageWrapper
      layout={layout}
      src={src}
      className="display-artwork__image"
      width={width}
      height={height}
      alt={artwork.title}
      index={artwork.index}
    />
  )
  if (linked) {
    return (
      <ArtworkImageLink href={"/artwork/" + artwork.slug}>
        {image}
      </ArtworkImageLink>
    )
  } else {
    return image
  }
}

ArtworkImage.defaultProps = {
  width: "100%",
  height: "auto",
}

const Artwork: React.SFC<ArtworkProps> = props => {
  return (
    <div className="display-artwork">
      <ArtworkImage {...props} />
      <ArtworkCaption {...props} />
    </div>
  )
}

Artwork.defaultProps = {
  linked: true,
}

const ArtworkImageLink = styled.a`
  text-decoration: none;
`

export default Artwork
