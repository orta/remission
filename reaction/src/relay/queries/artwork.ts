import * as Relay from "react-relay"

export default class ArtworkQueryConfig extends Relay.Route {
  public static queries = {
    artwork: (component, params) => Relay.QL`
      query {
        artwork(id: $artworkID) {
          ${component.getFragment("artwork", params)}
        }
      }
    `,
  }

  public static paramDefinitions = {
    artworkID: { required: true },
  }

  public static routeName = "ArtworkQueryConfig"
}
