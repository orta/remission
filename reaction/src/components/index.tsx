import Artwork from "./artwork"
import ArtworkFilter from "./artwork_filter"
import ArtworkGrid from "./artwork_grid"
import BorderedPulldown from "./bordered_pulldown"
import Buttons from "./buttons/index"
import Checkbox from "./checkbox"
import Gene from "./gene/index"
import Grid from "./grid"
import Icon from "./icon"
import Input from "./input"
import Modal from "./modal/modal"
import Nav from "./nav"
import Text from "./text"
import TextArea from "./text_area"
import TextLink from "./text_link"
import Title from "./title"

import * as React from "react"
import * as ReactDOM from "react-dom"
import * as Relay from "react-relay"

import { artsyNetworkLayer } from "../relay/config"
import * as Artsy from "./artsy"

interface InitOptions {
  user: User
  component: any
  domID: string
  queryConfig: any
}

export function init(options: InitOptions) {
  Relay.injectNetworkLayer(artsyNetworkLayer(options.user))

  const rootRoute = (
    <Artsy.ContextProvider currentUser={options.user}>
      <Relay.RootContainer Component={options.component} route={options.queryConfig} />
    </Artsy.ContextProvider>
  )

  ReactDOM.render(rootRoute, document.getElementById(options.domID))
}

export default {
  Artwork,
  ArtworkFilter,
  Buttons,
  Modal,
  ArtworkGrid,
  BorderedPulldown,
  Checkbox,
  Gene,
  Grid,
  Icon,
  Input,
  Nav,
  Text,
  TextArea,
  TextLink,
  Title,
}
