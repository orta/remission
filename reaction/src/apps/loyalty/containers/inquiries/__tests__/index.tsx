import * as React from "react"
import * as TestUtils from "react-dom/test-utils"
import * as renderer from "react-test-renderer"

import { Artwork } from "../../../../../components/artwork"
import { Inquiries, Props } from "../index"

const inquiryProps: Props = {
  user: {
    artwork_inquiries_connection: {
      edges: [
        {
          node: {
            id: "andy-warhol-skull",
            impulse_conversation_id: "1",
            artwork: {
              image: {
                url: "http://path.to.cat.pics",
              },
              artists: [
                {
                  __id: "percy-z",
                  name: "Percy Z",
                  href: "/percy-z",
                },
              ],
            },
          },
        },
        {
          node: {
            id: "andy-warhol-skullz",
            impulse_conversation_id: null,
            artwork: {
              image: {
                url: "http://path.to.cat.pics",
              },
              artists: [
                {
                  __id: "percy-z-convoless",
                  name: "Percy Z Without Conversation",
                  href: "/percy-z-without-conversation",
                },
              ],
            },
          },
        },
      ],
    },
  },
}

describe("inquiries", () => {
  it("renders the inquiries listing", () => {
    const inquiries = renderer.create(<Inquiries {...inquiryProps} />)
    expect(inquiries).toMatchSnapshot()
  })

  it("renders the artist information", () => {
    const inquiries = TestUtils.renderIntoDocument(<Inquiries {...inquiryProps} />) as Inquiries
    const aTags = TestUtils.scryRenderedDOMComponentsWithTag(inquiries, "a") as HTMLAnchorElement[]
    const artistLink = aTags.find(tag => tag.href === "/percy-z" && tag.textContent === "Percy Z")
    expect(artistLink).toBeTruthy()
    const noConvoLink = aTags.find(tag => tag.href === "/percy-z-without-conversation")
    expect(noConvoLink).toBeUndefined()
  })

  describe("concerning the submit button", () => {
    let inquiries: Inquiries
    let button: HTMLButtonElement

    beforeEach(() => {
      inquiries = TestUtils.renderIntoDocument(<Inquiries {...inquiryProps} />) as Inquiries
      button = TestUtils.findRenderedDOMComponentWithTag(inquiries, "button") as HTMLButtonElement
    })

    it("disables the submit button by default", () => {
      expect(button.disabled).toEqual(true)
    })

    it("enables the submit button when text is entered into the free-form textarea", () => {
      const textArea = TestUtils.findRenderedDOMComponentWithTag(inquiries, "textarea") as HTMLTextAreaElement

      textArea.value = "Mona Lisa, Don’t recall the dude’s name, Louvre Gift Shop"
      TestUtils.Simulate.change(textArea)
      expect(button.disabled).toEqual(false)

      textArea.value = " \n"
      TestUtils.Simulate.change(textArea)
      expect(button.disabled).toEqual(true)
    })

    it("enables the submit button when an inquiry is selected", () => {
      const artwork = TestUtils.findRenderedComponentWithType(inquiries, Artwork)
      const container = TestUtils.scryRenderedDOMComponentsWithTag(artwork, "div")[0]

      TestUtils.Simulate.click(container)
      expect(button.disabled).toEqual(false)

      TestUtils.Simulate.click(container)
      expect(button.disabled).toEqual(true)
    })
  })
})
