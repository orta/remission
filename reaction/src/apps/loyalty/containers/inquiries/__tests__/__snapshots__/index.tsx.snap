// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inquiries renders the inquiries listing 1`] = `
<div
  className="file__Container-cpd21h-1 ghYJhK"
>
  <div
    className="file__Nav-s1w500ep-0 krvdox"
  >
    <a
      className="file__NavIcon-s1w500ep-1 eNnazU"
      href="/"
    >
      <div
        className="file-s1ekx40u-0 krLqUx"
        style={undefined}
      >
        
      </div>
    </a>
    <a
      className="file__StyledLink-mgasw5-0 dIJcvg"
      href="/"
    >
      Back To Artsy
    </a>
  </div>
  <header
    className="file__Header-cpd21h-2 gEkXMR"
  >
    <div
      className="header-title file__StyledTitle-s3pz4jh-0 jziesf"
    >
      Please select all works you purchased
    </div>
    <p
      className="header-subtitle file__Text-s4rtu63-0 fjtRsH"
    >
      We will confirm submitted purchases with the galleries in order to qualify you for the program membership.
    </p>
  </header>
  <div
    className="artworks"
  >
    <div
      className="Grid__Grid-bojwg gdLArs"
    >
      <div
        className="file__Row-s12jglgi-1 kZldUL Row__Row-kNtYPA hicFUP"
      >
        <div
          className="file__Col-s12jglgi-0 jydIWb Col__Col-hCQanm jItcmX"
        >
          <div
            className="file__InquiryContainer-cpd21h-0 ghmnBd"
          >
            <div
              className="file__Container-s1284ws5-0 kvlZnW"
              onClick={[Function]}
              size={250}
            >
              <div
                className="file__ImageContainer-s1284ws5-1 jtqwav"
                size={250}
              >
                <img
                  className="file__Image-s1284ws5-2 kDvufT"
                  size={250}
                  src="http://path.to.cat.pics"
                />
                <div
                  className="overlay-container hovered"
                >
                  <div
                    className="file__OverlayBackground-sjjgzh-0 dqoUxd"
                  >
                    <div
                      className="file__Circle-sjjgzh-1 hSzDOv"
                      selected={false}
                    >
                      <div
                        className="file-s1ekx40u-0 bDXcCT"
                        style={undefined}
                      >
                        
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="file__StyledMetadata-s1l3srti-0 iIhayI"
              >
                <div>
                  <div
                    className="file__TruncatedLine-bzokla-0 dATVjh"
                  >
                    <strong>
                      <a
                        className="file__StyledTextLink-s18afosz-0 hlqDRc"
                        href="/percy-z"
                      >
                        Percy Z
                      </a>
                    </strong>
                  </div>
                  <div
                    className="file__TruncatedLine-bzokla-0 dATVjh"
                  >
                    <a
                      className="file__StyledTextLink-s18afosz-0 hlqDRc"
                      href={undefined}
                    >
                      <em />
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <p
      className="file__Text-s4rtu63-0 eEhqjQ"
    >
      If you purchased any works not included
      <br />
       above, please list them.
    </p>
    <textarea
      className="file__LargeTextArea-cpd21h-3 dFDpRL file-gpzlm6-0 iokafB"
      onChange={[Function]}
      placeholder="Artwork, Artist, Gallery"
    />
    <button
      className="file__InvertedButton-s2qkmvo-0 hmYmGS file__StyledButton-muis26-0 kaBVMO"
      disabled={true}
      onClick={[Function]}
    >
      <span>
        Submit purchases
      </span>
    </button>
  </footer>
</div>
`;
